enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTestLogic.java

```diff
@@ -689,6 +694,96 @@ public void shouldAllowAddingAndRemovingUserFromMultipleRoles() throws Exception
+    }
+
+    @Test
+    public void shouldNotCreateRoleIfInvalidRoleName() throws Exception
+    {
+        assertFail( adminSubject, "CALL dbms.security.createRole('')", "The provided role name is empty." );
+        assertFail( adminSubject, "CALL dbms.security.createRole('&%ss!')",
+                "Role name '&%ss!' contains illegal characters. Use simple ascii characters and numbers." );
+        assertFail( adminSubject, "CALL dbms.security.createRole('åäöø')",
+                "Role name 'åäöø' contains illegal characters. Use simple ascii characters and numbers" );
+    }
+
+    @Test
+    public void shouldNotCreateExistingRole() throws Exception
+    {
+        assertFail( adminSubject, format( "CALL dbms.security.createRole('%s')", ARCHITECT),
+                "The specified role 'architect' already exists" );
+        assertEmpty( adminSubject, "CALL dbms.security.createRole('new_role')" );
+        assertFail( adminSubject, "CALL dbms.security.createRole('new_role')",
+                "The specified role 'new_role' already exists" );
+    }
+
+    @Test
+    public void shouldNotAllowNonAdminCreateRole() throws Exception
+    {
+        testFailCreateRole( pwdSubject, CHANGE_PWD_ERR_MSG );
+        testFailCreateRole( readSubject, PERMISSION_DENIED );
+        testFailCreateRole( writeSubject, PERMISSION_DENIED );
+        testFailCreateRole( schemaSubject, PERMISSION_DENIED );
+    }
+
+    //---------- delete role -----------
+
+    @Test
+    public void shouldThrowIfTryingToDeletePredefinedRole() throws Exception
+    {
+        testFailDeleteRole( adminSubject, ADMIN,
+                format( "'%s' is a predefined role and can not be deleted.", ADMIN ) );
+        testFailDeleteRole( adminSubject, ARCHITECT,
+                format( "'%s' is a predefined role and can not be deleted.", ARCHITECT ) );
+        testFailDeleteRole( adminSubject, PUBLISHER,
+                format( "'%s' is a predefined role and can not be deleted.", PUBLISHER ) );
+        testFailDeleteRole( adminSubject, READER,
+                format( "'%s' is a predefined role and can not be deleted.", READER ) );
+    }
+
+    @Test
+    public void shouldThrowIfNonAdminTryingToDeleteRole() throws Exception
+    {
+        assertEmpty( adminSubject, format("CALL dbms.security.createRole('%s')", "new_role" ) );
+        testFailDeleteRole( schemaSubject, "new_role", PERMISSION_DENIED);
+        testFailDeleteRole( writeSubject, "new_role", PERMISSION_DENIED);
+        testFailDeleteRole( readSubject, "new_role", PERMISSION_DENIED);
+        testFailDeleteRole( noneSubject, "new_role", PERMISSION_DENIED);
+    }
+
+    @Test
+    public void shouldThrowIfDeletingNonExistentRole()
+    {
+        testFailDeleteRole( adminSubject, "nonExistent", "Role 'nonExistent' does not exist." );
+    }
+
+    @Test
+    public void shouldDeleteRole() throws Exception
+    {
+        assertEmpty( adminSubject, format("CALL dbms.security.createRole('%s')", "new_role") );
```

Comment:
Here one could use `NeoInteractionLevel.getManager` to get a backdoor for creating the role without the full cypher call through the interactionlevel